### PR TITLE
feat(create-mud): use solhint-plugin-mud in templates

### PR DESCRIPTION
**Pull request by dk1a**
needs the plugin to be published to work correctly

### CODE REVIEW

**Comment by holic** [comment]
if you add this, it should release automatically via CI
https://github.com/latticexyz/mud/blob/d38232032b86d3dbccc00ffc23f6eee4e93c7732/packages/dev-tools/package.json#L55-L57

File: packages/solhint-plugin-mud/package.json [hunk 1]
```diff
@@ -23,5 +23,8 @@
   "devDependencies": {
     "tsup": "^6.7.0"
   },
+  "publishConfig": {
+    "access": "public"
+  },
```
Comment (dk1a) [inline packages/solhint-plugin-mud/package.json:28]: @holic did you mean here?

Comment (holic) [reply packages/solhint-plugin-mud/package.json:28]: yep! though our linter will prob complain about exact placement in this, might wanna run `pnpm sort-package-json` from root

File: packages/create-mud/scripts/copy-templates.sh [hunk 2]
```diff
@@ -11,6 +11,7 @@ find ./dist/templates/* -name "package.json" -type f | while read -r file; do
   cat "$file"
   # GPT-4 recommended perl to edit-in-place rather than sed, because sed wasn't working on CI
   perl -pi -e 's|"(?=@latticexyz)([^"]+)":\s*"link:[^"]+"|"\1": "{{mud-version}}"|g' "$file"
+  perl -pi -e 's|"(solhint-plugin-mud)":\s*"file:[^"]+"|"\1": "{{mud-version}}"|g' "$file"
```
Comment (holic) [inline packages/create-mud/scripts/copy-templates.sh:14]: just to confirm this works as expected, you can run `pnpm build` in `create-mud` and look at `dist/templates` to see if the solhint plugin version got swapped with the latest canary?

Comment (dk1a) [reply packages/create-mud/scripts/copy-templates.sh:14]: Capture

File: packages/solhint-plugin-mud/package.json (hunk 1 above)
Comment (dk1a) [reply packages/solhint-plugin-mud/package.json:28]: linter seems to think this is the correct order

**Review by holic — APPROVED** [review]

**Pull request merged**